src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java

```diff
@@ -86,23 +91,59 @@ protected static Configuration getConfiguration() throws CheckstyleException {
     }
 
     /**
-     * Creates {@link DefaultConfiguration} instance for the given check class.
-     * @param clazz check class.
+     * Creates {@link DefaultConfiguration} instance for the given module class.
+     * @param clazz module class.
      * @return {@link DefaultConfiguration} instance.
      */
-    private static DefaultConfiguration createCheckConfig(Class<?> clazz) {
+    private static DefaultConfiguration createModuleConfig(Class<?> clazz) {
         return new DefaultConfiguration(clazz.getName());
     }
 
     /**
+     * Creates {@link Checker} instance based on the given {@link Configuration} instance.
+     * @param moduleConfig {@link Configuration} instance.
+     * @return {@link Checker} instance based on the given {@link Configuration} instance.
+     * @throws Exception if an exception occurs during checker configuration.
+     */
+    public Checker createChecker(Configuration moduleConfig)
+            throws Exception {
+        if (checkstyleModules == null) {
+            checkstyleModules = CheckUtil.getCheckstyleModules();
+        }
+
+        final String name = moduleConfig.getName();
+        boolean addTreeWalker = false;
+
+        for (Class<?> moduleClass : checkstyleModules) {
+            if (moduleClass.getName().equals(name)) {
+                if (ModuleReflectionUtils.isCheckstyleCheck(moduleClass)
+                        || ModuleReflectionUtils.isTreeWalkerFilterModule(moduleClass)) {
+                    addTreeWalker = true;
+                }
+            }
+        }
+
+        return createChecker(moduleConfig, addTreeWalker);
+    }
+
+    /**
      * Creates {@link Checker} instance based on specified {@link Configuration}.
-     * @param checkConfig {@link Configuration} instance.
+     * @param moduleConfig {@link Configuration} instance.
      * @return {@link Checker} instance.
      * @throws CheckstyleException if an exception occurs during checker configuration.
+     * @noinspection BooleanParameter
```

Comment:
@romani This noinspection was added because changes for `test` PR haven't been finalized.
We can add it to that issue, or wait for that PR to be merged and I will fix it here.